src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -22,6 +22,8 @@
 
   <PropertyGroup>
     <TestWithCore Condition="'$(TestWithCore)' == '' And '$(TestTFM)' != 'netcoreapp1.0'">false</TestWithCore>
+    <!-- TestAppX is true only for debug scenario which runs on non-aot runtime ids -->
+    <TestAppX Condition="'$(TestWithCore)' == 'false' And '$(TestTFM)' == 'netcore50' And !$(TestNugetRuntimeId.Contains('aot'))">true</TestAppX>
```

Comment:
So here TestWithCore is used solely to set the TestHostExecutable to CoreRun.exe. In the case of netcore50 we use the xunit.console.uwp.exe which is not a Core app, but a console app that installs the UWP test app.